sycl/source/detail/xpti_registry.hpp

```diff
@@ -178,17 +178,12 @@ class XPTIScope {
         MTraceType(0) {
     detail::tls_code_loc_t Tls;
     auto TData = Tls.query();
-    // If TLS is not set, we can still genertate universal IDs with user data
-    // and CodePtr information
-    const char *FuncName = UserData;
-    if (TData.functionName())
-      FuncName = TData.functionName();
     // Create a tracepoint object that has a lifetime of this class
-    MTP = new TracePoint(TData.fileName(), FuncName, TData.lineNumber(),
-                         TData.columnNumber(), CodePtr);
-    if (MTraceType == (uint16_t)xpti::trace_point_type_t::graph_create ||
-        MTraceType == (uint16_t)xpti::trace_point_type_t::node_create ||
-        MTraceType == (uint16_t)xpti::trace_point_type_t::edge_create)
+    MTP = new TracePoint(TData.fileName(), TData.functionName(),
```

Comment:
Since the contract regarding code_location has been changed, this will work for this use-case.  We may need to look at how code_location is being used and may have to address it for all cases.

Context:
- Previous code_location had all attributes set to NULL when NDEBUG was used and the previous code here handled it.
- New definition of code_location only zeroes out file_location when NDEBUG is used, which changed the contract code_location had with all previous use

Reply:
"may have to address it for all cases."
do you mean to handle the case with empty code_location data? Please correct me if I am wrong, I didn't understand why UserData usage is better there because it will be the same for all queue.memset calls and will not bring any benefits. If code_location is not available the only hope to distinguish calls is CodePtr which is still used, doesn't it?

Reply:
As long as the metadata includes the node_create was for memset or memcpy, we are good.

Reply:
@tovinkere got it, thanks for the clarification. I walked through the SYCL RT code and it looks like we do not have any code_location absence WAs left.